targets/wasm_exec.js

```diff
@@ -466,20 +466,13 @@
 			this._idPool = [];      // unused ids that have been garbage collected
 			this.exited = false;    // whether the Go program has exited
 
-			while (true) {
-				const callbackPromise = new Promise((resolve) => {
-					this._resolveCallbackPromise = () => {
-						if (this.exited) {
-							throw new Error("bad callback: Go program has already exited");
-						}
-						setTimeout(resolve, 0); // make sure it is asynchronous
-					};
-				});
+			if (this._inst.exports._start) {
 				this._inst.exports._start();
-				if (this.exited) {
-					break;
-				}
-				await callbackPromise;
+
+				// TODO: wait until the program exists.
```

Comment:
Seems important?

Reply:
The behavior should be the same as it was before. If you read the code carefully, you see that it wait for `callbackPromise`, which is resolved when `this._resolveCallbackPromise` is called, but `this._resolveCallbackPromise` is never called.
So `await new Promise(() => {});` should be exactly the same behavior: never resolving. It basically just simplifies the code.